app/Http/Controllers/Api/GeneralController.php

```diff
@@ -44,4 +46,48 @@ public function version()
             'latest'    => $latest,
         ])->item(CACHET_VERSION);
     }
+
+    /**
+     * Show the general system status.
+     *
+     * @return \Illuminate\Http\JsonResponse
+     */
+    public function status()
+    {
+        $totalComponents = Component::enabled()->count();
+        $majorOutages = Component::enabled()->status(4)->count();
+        $isMajorOutage = $totalComponents ? ($majorOutages / $totalComponents) >= 0.5 : false;
+
+        $withData = [
+            'system_status'  => 'info',
+            'system_message' => trans_choice('cachet.service.bad', $totalComponents),
+            'favicon'        => 'favicon-high-alert',
+        ];
+
+        if ($isMajorOutage) {
+            $withData = [
+                'system_status'  => 'danger',
+                'system_message' => trans_choice('cachet.service.major', $totalComponents),
+                'favicon'        => 'favicon-high-alert',
+            ];
+        } elseif (Component::enabled()->notStatus(1)->count() === 0) {
+            // If all our components are ok, do we have any non-fixed incidents?
+            $incidents = Incident::notScheduled()->orderBy('created_at', 'desc')->get();
+            $incidentCount = $incidents->count();
+
+            if ($incidentCount === 0 || ($incidentCount >= 1 && (int) $incidents->first()->status === 4)) {
+                $withData = [
+                    'system_status'  => 'success',
+                    'system_message' => trans_choice('cachet.service.good', $totalComponents),
+                    'favicon'        => 'favicon',
```

Comment:
Yeah, on the page: https://social.ringplus.net/categories/community there's a section that checks a cachet status page and just displays the same exact message the cachet status page displays, on the backend of that site it gets all components and loops through them to see if any of them are not functional which then displays the message from there. It'll be better if I can just use this API endpoint to get the message directly. Also check the Issue #1936